src/Type.zig

```diff
@@ -890,7 +890,7 @@ pub fn alignof(ty: Type, comp: *const Compilation) u29 {
         .@"enum" => if (ty.data.@"enum".isIncomplete() and !ty.data.@"enum".fixed) 0 else ty.data.@"enum".tag_ty.alignof(comp),
         .typeof_type, .decayed_typeof_type => ty.data.sub_type.alignof(comp),
         .typeof_expr, .decayed_typeof_expr => ty.data.expr.ty.alignof(comp),
-        .attributed => ty.data.attributed.base.alignof(comp),
+        .attributed => ty.requestedAlignment(comp) orelse ty.data.attributed.base.alignof(comp),
```

Comment:
In this case it is a record (attributed record). As far as I can tell a couple things are going on: `__attribute__((aligned(N)))` directly on a record overrides the record's computed alignment, unless it's being asked to decrease the alignment. Putting the attribute on a typedef, however, *can* decrease the alignment. So in the example you just posted, the attribute is on the anonymous struct, which is being typedef'ed to the name `X`. Since the attribute is directly on the struct, it can't reduce the alignment to 1.

Reply:
I see.

But you can tell if the attributes are in front of or behind the typedef, and respond accordingly? So if `Type` looks like `.attribute -> .typeof_type -> .record` you return the attribute. If it looks like `.typeof_type -> .attribute -> .record` you return the computed record alignment.

Reply:
`typeof_type` is for `typeof(X)`, not typedefs. The problem is that, conceptually, `alignof()` a record type needs to consider attributes if it's a typedef'ed type, and not consider them if it's not. But I don't think that info is currently available at the time `alignof` is called.

Reply:
I also think it's compiler-dependent. For MSVC it looks like decels before the typedef are "for" the record, not the type.